lib/array_equals.rb

```diff
@@ -1,5 +1,18 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == nil && array2 == nil || array1 == [] && array2 == []
```

Comment:
Good boundary checking! 

Style nit: this many `if`-statements all in one line is a bit confusing. (You have to know Ruby's order of operations in order to understand it. :smile: )
- Good: use parenthesis around the "and" clauses (e.g. `if (array1 == nil && array2 == nil) || (array1 == [] && array2 == [])`)
- Better: break up each condition (e.g. `== nil` and `== []`) into separate "guarding" `if`-statements

e.g.
```
# Null/empty checks
if array1 == nil && array2 == nil
  return true
elsif array1 == [] && array2 == []
  return true
elsif array1 == nil || array2 == nil
  return false

# Length check
if array1.length != array2.length
  return false
end

...
```

You _might_ even be able to do this - but some people would say this
is not explicit enough (or "has too much magic")
```
if array1 == array2:
  return True
elsif array1 == nil || array2 == nil:
  return False
elsif array1 == [] || array2 == []:
  return False
end
```